src/Sylius/Bundle/CoreBundle/OAuth/UserProvider.php

```diff
@@ -132,17 +141,29 @@ private function createUserByOAuthUserResponse(UserResponseInterface $response)
     {
         /** @var SyliusUserInterface $user */
         $user = $this->userFactory->createNew();
-        /** @var CustomerInterface $customer */
-        $customer = $this->customerFactory->createNew();
+
+        $customer = $this->customerRepository->findOneBy(['emailCanonical' => strtolower($response->getEmail())]);
```

Comment:
Instead of `strtolower` we should use canonicalizer service, which is injected already to have coherent behaviour.